src/parser/shaman/elemental/modules/talents/Ascendance.js

```diff
@@ -35,36 +52,94 @@ class Ascendance extends Analyzer {
     return this.rawUpdate;
   }
 
+  get averageLavaBurstCasts() {
+    return this.numCasts[SPELLS.LAVA_BURST.id]/this.numCasts[SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id];
+  }
+
   on_byPlayer_cast(event) {
-    if (this.selectedCombatant.hasBuff(SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id, event.timestamp)) {
-      const spellId = event.ability.guid;
-      if (this.numCasts[spellId] !== undefined) {
-        this.numCasts[spellId] += 1;
-      } else {
-        this.numCasts.others += 1;
+    const spellId = event.ability.guid;
+    const target = this.enemies.getEntity(event);
+
+
+    if(spellId === SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id) {
+      this.justEnteredAcendance = true;
+      this.numCasts[SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id]++;
```

Comment:
We don't use increments, please use += 1
```suggestion
      this.numCasts[SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id] += 1;
```